rocAL/source/pipeline/pipeline_serializer.cpp

```diff
@@ -145,6 +147,16 @@ void serialize_parameter_by_type(rocal_proto::Parameter *parameter, const Argume
 
 // Main function with improved structure and error handling
 void serialize_parameter_to_protobuf(rocal_proto::Parameter *parameter, const Argument &op_arg) {
```

Comment:
[Non-blocking] The new `sub_type_name` → `RocalParameterType` mapping in `serialize_parameter_to_protobuf()` is string-based and throws if the underlying parameter core does not match the expected subclass. That keeps serialization honest, but also makes this path a bit brittle to new parameter subclasses or typos in `sub_type_name`. Longer term it might be safer to centralize the mapping from `sub_type_name` to `RocalParameterType` (or an enum) so new parameter types can plug in without needing to touch this helper.